testsuite/features/step_definitions/command_steps.rb

```diff
@@ -1123,10 +1123,24 @@
 
   # Actually define the VM, but don't start it
   raise 'not found: virt-install' unless file_exists?(node, '/usr/bin/virt-install')
-  node.run("virt-install --name #{vm_name} --memory 512 --vcpus 1 --disk path=#{disk_path} "\
-           "--network network=test-net0 --graphics vnc,listen=0.0.0.0 "\
-           "--serial file,path=/tmp/#{vm_name}.console.log "\
-           "--import --hvm --noautoconsole --noreboot")
+  case host
+  when 'kvm_server'
+    # use virtio bus for KVM
+    node.run(
+      "virt-install --name #{vm_name} --memory 512 --vcpus 1 --disk path=#{disk_path},bus=virtio "\
+      "--network network=test-net0 --graphics vnc,listen=0.0.0.0 "\
+      "--serial file,path=/tmp/#{vm_name}.console.log "\
+      "--import --hvm --noautoconsole --noreboot --osinfo sle15sp4"
+    )
+  when 'xen_server'
+    # Use ide bus for Xen
+    node.run(
+      "virt-install --name #{vm_name} --memory 512 --vcpus 1 --disk path=#{disk_path},bus=ide "\
+      "--network network=test-net0 --graphics vnc,listen=0.0.0.0 "\
+      "--serial file,path=/tmp/#{vm_name}.console.log "\
+      "--import --hvm --noautoconsole --noreboot --osinfo sle15sp4"
+    )
+  end
```

Comment:
Wow, all this duplicated code just for this piece of code? `bus=#{bus_type}`

Reply:
can't we just use a bus variable and name it according to the case? :thinking:

Reply:
something like:
```suggestion
  case host
  when 'kvm_server'
    # use virtio bus for KVM
     bus = 'virtio'
  when 'xen_server'
     # Use ide bus for Xen
     bus = 'ide'
  end
  node.run(
    "virt-install --name #{vm_name} --memory 512 --vcpus 1 --disk path=#{disk_path},bus=#{bus} "\
      "--network network=test-net0 --graphics vnc,listen=0.0.0.0 "\
      "--serial file,path=/tmp/#{vm_name}.console.log "\
      "--import --hvm --noautoconsole --noreboot --osinfo sle15sp4"
  )
end
```

Reply:
Fixed.